### DISCUSSION TITLE
建议使用其他云存储服务存储视频文件

### DISCUSSION THREAD

**Discussion by Tseing** [Nang-muoi-sing/cdo-corpus#35] [Ideas]
视频文件体积较大，不适合存储在 Github 仓库，因为 `git clone` 和 `git pull` 的速度过慢，常常导致连接失败，这不太便于工作。建议使用坚果云、OneDrive 等云存储服务存储视频文件，只需在 Markdown 中列出每个视频的标题及分享链接，协作者需要哪个视频就下载哪个视频，这样会方便许多。

**Comment by Guanchishan** [comment]
好主意。不过即便将视频文件迁移走后，他会否仍然残留在git文件中占用大小？

**Reply by Tseing** [reply]
的确如此，`.git` 文件包含每次 commit 的信息。有两种解决方案，
- 其一是使用 `filter-branch` 删除 `.git` 文件中的视频文件，这种方法一劳永逸，直接减小文件体积，但这种方法操作不当会引起比较严重的后果，造成文件无法回退等。
- 另一种方法比较简单，直接将文件夹中的视频、音频文件删除后 push 到仓库，不改动历史 commit 信息。需要提醒协作者在以后的初次 pull 要使用 `git pull --depth 1`，只拉取最近的一次更新，就也不会下载大体积的视频文件了。

**Reply by Guanchishan** [reply]
感谢建议。我另外弄个分支或repo尝试下第一种方法。成功以后再于此repo的此分支使用之。

最近忙于尘务，多谢你联络！之后聊聊如何就本项目协作。

大方向上我也很赞同将视频音频文件从本repo解开。优先级是先视频后音频。音频似乎占地没那么大，解不解无所谓。

**Reply by Tseing** [reply]
好的，我也刚刚脱身，欢迎联络 : )

**Comment by Guanchishan** [comment]
感谢建议。我另外弄个分支或repo尝试下第一种方法。成功以后再于此repo的此分支使用之。

最近忙于尘务，多谢你联络！之后聊聊如何协作。

Leo ***@***.***> 于2022年8月13日周六 18:08写道：
